src/redux/reducers/todoReducer.js

```diff
@@ -0,0 +1,56 @@
+import * as types from '../types';
+
+const initialState = {
+    todosData: [],
+};
+
+const saveToLocalStorage = function (state) {
+    localStorage.setItem('todos', JSON.stringify(state));
+}
+
+const retrieveFromLocalStorage = function () {
+    return JSON.parse(localStorage.getItem('todos'))
+}
+
+export default function (state = initialState, action) {
+    let newState;
+    switch (action.type) {
+        case types.FETCH_TODOS:
+            const retrievedState = retrieveFromLocalStorage() || initialState
+            newState = { ...retrievedState }
+            return newState
+
+        case types.ADD_TODO:
+            newState = { ...state, todosData: state.todosData.concat({ id: state.todosData.length, title: action.payload, isCompleted: false }) }
+            saveToLocalStorage(newState)
```

Comment:
It's better to subscribe to store changes and save state to local storage with any change rather than calling the same function in each separate action